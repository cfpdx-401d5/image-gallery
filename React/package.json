{
  "name": "image-gallery",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "body-parser": "^1.17.1",
    "cors": "^2.8.1",
    "express": "^4.15.2",
    "mongoose": "^4.8.7",
    "morgan": "^1.8.1",
    "react": "^15.4.2",
    "react-dom": "^15.4.2"
  },
  "devDependencies": {
    "chai": "^3.5.0",
    "chai-http": "^3.0.0",
    "enzyme": "^2.7.1",
    "enzyme-to-json": "^1.5.0",
    "eslint": "^3.17.1",
    "eslint-plugin-react": "^6.10.0",
    "mocha": "^3.2.0",
    "nodemon": "^1.11.0",
    "react-addons-test-utils": "^15.4.2",
    "react-scripts": "0.9.4"
  },
  "scripts": {
    "start": "react-scripts start",
    "start-server": "nodemon server.js",
    "build": "react-scripts build",
    "test": "react-scripts test --env=jsdom",
    "eject": "react-scripts eject"
  }
}
