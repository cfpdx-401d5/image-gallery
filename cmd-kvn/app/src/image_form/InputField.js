import React, { PropTypes } from 'react';

function InputField(props) {
  return (
    <input
      name={props.name}
      type='text'
      placeholder={`Enter ${props.name}`}
      defaultValue=''
      onChange={props.handler}
    />
  );
}

InputField.propTypes = {
  name: PropTypes.string.isRequired,
  handler: PropTypes.func.isRequired,
};

export default InputField;
