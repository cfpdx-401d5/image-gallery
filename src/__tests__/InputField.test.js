/* eslint-disable */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import InputField from '../image_form/InputField';

describe('InputField component', function () {
    it('renders the InputField component', () => {
        const rendered = shallow(<InputField name='_name' handler={() => {}} />);
        expect(toJson(rendered)).toMatchSnapshot();
    });
});
