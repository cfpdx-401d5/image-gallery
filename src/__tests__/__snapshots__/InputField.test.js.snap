exports[`InputField component renders the InputField component 1`] = `
<input
  defaultValue=""
  name="_name"
  onChange={[Function]}
  placeholder="Enter _name"
  type="text" />
`;
